feat: Add playground to website

phated opened the pull request:
It's time for @spotandjake's wildest dreams to come true.

Casually ignore the committed 12mb grainc.bc.mjs - don't have a good way to automatically pull that into the repo.

marcusroberts commented:
When I try the preview build, it compiles and runs fine, but any changes I make to the source are then ignored in future compiles

phated commented:
> When I try the preview build, it compiles and runs fine, but any changes I make to the source are then ignored in future compiles

This is not enough information to explain what you are doing (or doing wrong). I can say for a fact that I developed everything with `npm run start-playground` and the rebuilds work perfectly.

marcusroberts commented:
When I click the Deploy Preview link to Netlify in his PR. (https://deploy-preview-316--grain-lang.netlify.app/playground) and press Compile and Run, it works fine, but any changes I make in the source code panel aren't reflected when I press Compile and Run a second time.

phated commented:
> When I click the Deploy Preview link to Netlify in this PR. (https://deploy-preview-316--grain-lang.netlify.app/playground) and press Compile and Run, it works fine, but any changes I make in the source code panel aren't reflected when I press Compile and Run a second time.

There is a bug in the js-runner that @ospencer needs to look into that causes it to be stateful and break multiple runs.

phated commented:
@marcusroberts I dropped Grain's js-runner completely and switched to using the new 1.x version of `@wasmer/wasi` directly.

I tried a few simple examples but can you test your stuff again once the new deploy finishes?

marcusroberts commented:
That fixed it!

marcusroberts reviewed (approved):
Works great for me!

ospencer reviewed (approved):
Excellent excellent work!

phated commented on .gitignore:
```diff
@@ -8,3 +8,5 @@ _multiconfig.yml
 /docs
 /guide
 /blog
+/playground
```
```suggestion
/try
```

phated commented on vite.config.js:
```diff
@@ -0,0 +1,44 @@
+module.exports = ({ mode }) => {
+  return {
+    root,
+    base: mode === "development" ? "/" : "/playground/",
```
```suggestion
    base: mode === "development" ? "/" : "/try/",
```

phated commented on vite.config.js:
```diff
@@ -0,0 +1,44 @@
+
+    build: {
+      target: "es2020",
+      outDir: "../playground",
```
```suggestion
      outDir: "../try",
```

phated reviewed (commented):
Talked with @ospencer and I think we want this URL to be `/try`. These changes should do that

phated commented on index.html:
```diff
@@ -52,6 +52,7 @@
         <a href="/docs/intro" class="nav-link heading">Documentation</a>
         <a href="https://github.com/grain-lang/grain/graphs/contributors" class="nav-link heading">Community</a>
         <a href="/blog" class="nav-link heading">Blog</a>
+        <a href="/playground" class="nav-link heading">Try</a>
```
```suggestion
        <a href="/try" class="nav-link heading">Try</a>
```

phated commented on index.html:
```diff
@@ -64,6 +65,7 @@
       <a href="/docs/intro" class="nav-link heading">Documentation</a>
       <a href="https://github.com/grain-lang/grain/graphs/contributors" class="nav-link heading">Community</a>
       <a href="/blog" class="nav-link heading">Blog</a>
+      <a href="/playground" class="nav-link heading">Try</a>
```
```suggestion
      <a href="/try" class="nav-link heading">Try</a>
```

phated commented on playground_src/index.html:
```diff
@@ -0,0 +1,89 @@
+                <a href="/docs/intro" class="nav-link heading">Documentation</a>
+                <a href="https://github.com/grain-lang/grain/graphs/contributors" class="nav-link heading">Community</a>
+                <a href="/blog" class="nav-link heading">Blog</a>
+                <a href="/playground" class="nav-link heading">Try</a>
```
```suggestion
                <a href="/try" class="nav-link heading">Try</a>
```

phated commented on playground_src/index.html:
```diff
@@ -0,0 +1,89 @@
+            <a href="/docs/intro" class="nav-link heading">Documentation</a>
+            <a href="https://github.com/grain-lang/grain/graphs/contributors" class="nav-link heading">Community</a>
+            <a href="/blog" class="nav-link heading">Blog</a>
+            <a href="/playground" class="nav-link heading">Try</a>
```
```suggestion
            <a href="/try" class="nav-link heading">Try</a>
```

phated commented on themes/grain/layout/_partial/hero.ejs:
```diff
@@ -13,6 +13,7 @@
       <a href="/docs/intro" class="nav-link heading">Documentation</a>
       <a href="https://github.com/grain-lang/grain/graphs/contributors" class="nav-link heading">Community</a>
       <a href="/blog" class="nav-link heading">Blog</a>
+      <a href="/playground" class="nav-link heading">Try</a>
```
```suggestion
      <a href="/try" class="nav-link heading">Try</a>
```
